### PR TITLE
Add runtime configuration & upgrade CI build

### PR DESCRIPTION
**Pull request by jstorm31**
- Added runtime (server) configuration with Configuru
- Documented configuration
- Upgraded Docker base image
- Added Dockerfile and .dockerignore
- Upgraded node to version 18.12 (which is now stable)

### CODE REVIEW

**Review by horaklukas — APPROVED** [review]

**Review by horaklukas — APPROVED** [review]
Seems good to me 👍

**Comment by jstorm31** [comment]
Thank you for following. I'll just add one more change with server configuration from Secrets Manager later 🙂

**Comment by jstorm31** [comment]
I added configuration with Configuru package and also upgraded node. Could you please re-review @horaklukas ? 🙏

File: template/.nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+18.12
```
Comment (horaklukas) [inline template/.nvmrc:1]: Nice 👍

File: template/src/config/serverConfig.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+console.log(`Loading the server configuration from ${process.env.CFG_JSON_PATH}`);
+
+const loader = createLoader({
+    defaultConfigPath: '.env.jsonc',
+});
+
+export const serverConfig = values({
+    // TODO: Add your server configuration here
+    myEnvVar: loader.string('MY_ENV_VAR'),
```
Comment (horaklukas) [inline template/src/config/serverConfig.ts:11]: do we need add all variables we add to the `.env.local.jsonc` also here? If sou it worths to mention it in the readme..

Comment (jstorm31) [reply template/src/config/serverConfig.ts:11]: Yes, here it's a mapping to the config object and adding type safety. Good point, I'll add it to the readme👍

Comment (jstorm31) [reply template/src/config/serverConfig.ts:11]: I updated the docs 📝

File: template/.nvmrc (hunk 1 above)
Comment (horaklukas) [inline template/.nvmrc:1]: @jstorm31 did you try this? It seems like `madge` has `detective-postcss` dependency which requires at most node 16...

Comment (horaklukas) [reply template/.nvmrc:1]: https://github.com/pahen/madge/issues/329 :/

Comment (jstorm31) [reply template/.nvmrc:1]: Sad story, downgraded back to version `16.19`

**Comment by jstorm31** [comment]
Merging this

**Pull request merged**